Moving release process to Bintray

langsmith opened the pull request:
Resolves #70 by moving the release/snapshot process to Bintray.

Related: 

- https://github.com/mapbox/mapbox-plugins-android/pull/961
- https://github.com/mapbox/mapbox-java/pull/1037

langsmith commented:
More weird Bintray 🐛 at the finish line of these Bintray prs. For this repo, it's a `Invalid publication 'MapboxGesturesPublication': artifact file does not exist: '/Users/LangstonSmith/Mapbox/mapbox-gestures-android/library/build/libs/library-sources.jar.asc'` message

On my local machine, `make release` works just fine for building the project. When I try to publish to Bintray via `make publish-stable`, I get the following 💥below:

<img width="1122" alt="Screen Shot 2019-07-01 at 10 18 27 AM" src="https://user-images.githubusercontent.com/4394910/60454883-df723580-9be9-11e9-9138-3f7fd83cdcce.png">

No worries if not @LukasPaczos , but any ideas about the missing `.asc` file?

langsmith commented:
Found https://github.com/gradle/gradle/issues/3635 and tried some suggestions in that thread, but nothing worked.

langsmith commented:
Also related:

https://github.com/asciidoctor/asciidoctorj/issues/470

https://discuss.gradle.org/t/how-to-publish-artifacts-signatures-asc-files-using-maven-publish-plugin/7422/23

https://github.com/bintray/gradle-bintray-plugin/issues/255#issuecomment-423678744 looks promising but that's with the assumption that the `.asc` file is even generated.

LukasPaczos commented:
> any ideas about the missing .asc file

I believe that those are the signed artifacts. Are you sure that your `mapboxGpgPassphrase` was correctly set on your local machine when you tested?

langsmith commented:
Figured out what the issue was. I hadn't deleted the `mvn-push-android.gradle` file and the reference to it at https://github.com/mapbox/mapbox-gestures-android/blob/master/build.gradle#L28 🤦‍♂🤦‍♂🤦‍♂🤦‍♂. It was interfering with my new Bintray setup .

When I deleted it, ran `make release`, and then ran `make publish-stable`, everything uploaded to Bintray just fine 🎊  https://bintray.com/mapbox/mapbox/mapbox-android-gestures

_As a test_ and to get MavenCentral sync permission, I've published 0.5.0 on Bintray. We can delete later.

![Screen Shot 2019-07-02 at 10 24 59 AM](https://user-images.githubusercontent.com/4394910/60533147-be761700-9cb3-11e9-8ca0-84bc10a192a9.png)

Still getting that weird message when I request SNAPSHOT, but we can cross that bridge when we get to it. Things seemed to resolve themselves in https://github.com/mapbox/mapbox-java/pull/1037
![Screen Shot 2019-07-02 at 10 28 23 AM](https://user-images.githubusercontent.com/4394910/60533336-38a69b80-9cb4-11e9-9b80-ac862e8a8997.png)

![Screen Shot 2019-07-02 at 10 28 55 AM](https://user-images.githubusercontent.com/4394910/60533335-38a69b80-9cb4-11e9-8f0a-68d8e5e019f0.png)

langsmith commented:
Looks like we're good to go with SNAPSHOTs too

<img width="979" alt="Screen Shot 2019-07-02 at 10 49 42 PM" src="https://user-images.githubusercontent.com/4394910/60566477-dc776200-9d1b-11e9-8264-8e147ba40a94.png">

LukasPaczos commented on Makefile:
```diff
@@ -11,9 +11,13 @@ test:
+publish-stable:
+	export IS_LOCAL_DEVELOPMENT=false; ./gradlew :library:bintrayUpload
 
 publish-local:
```
This does not work anymore and can be removed.

langsmith replied:
Removed

LukasPaczos commented on build.gradle:
```diff
@@ -5,10 +5,14 @@ buildscript {
         google()
         jcenter()
+        // For publishing files to Bintray
+        maven { url 'https://oss.jfrog.org/artifactory/oss-snapshot-local/' }
```
This line is not necessary to publish to arttifactory. Could you remove it?

langsmith replied:
Removed

LukasPaczos commented on circle.yml:
```diff
@@ -62,23 +62,27 @@ jobs:
           command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
-            make publish ;
+            if [[ $CIRCLE_BRANCH == master ]] || [[ $CIRCLE_BRANCH == release-v* ]] || [[ $CIRCLE_TAG == v* ]]; then
```
We are using the `v[number]` convention in this repo, so `[[ $CIRCLE_BRANCH == release-v* ]]` can be skipped.

langsmith replied:
Removed

LukasPaczos commented on gradle/javadoc.gradle:
```diff
@@ -0,0 +1,19 @@
+apply from: "${rootDir}/gradle/dependencies.gradle"
+
+android.libraryVariants.all { variant ->
```
This file means that `/library/javadoc.gradle` can be removed.

langsmith replied:
Removed

langsmith commented:
>The release process is missing the step that changes the version name and removes the SNAPSHOT reference in order to support a full tag release. Check out this step.

Added to `circle.yml`.
>Also, the readme should be updated with the new snapshot repository URL.

Updated.

langsmith commented:
Ok @LukasPaczos , just pushed a commit to address your latest review.

LukasPaczos commented on Makefile:
```diff
@@ -11,9 +11,9 @@ test:
-publish:
-	export IS_LOCAL_DEVELOPMENT=false; ./gradlew :library:uploadArchives
+publish-stable:
+	export IS_LOCAL_DEVELOPMENT=false; ./gradlew :library:bintrayUpload
```
`IS_LOCAL_DEVELOPMENT` is not used.

langsmith replied:
Removed

LukasPaczos commented on library/build.gradle:
```diff
@@ -1,4 +1,8 @@
 apply plugin: 'com.android.library'
+apply plugin: 'com.jfrog.bintray'
```
Already applied in the `gradle-bintray.gradle`.

LukasPaczos commented on library/build.gradle:
```diff
@@ -1,4 +1,8 @@
 apply plugin: 'com.android.library'
+apply plugin: 'com.jfrog.bintray'
+apply plugin: "com.jfrog.artifactory"
```
Same as above.

langsmith replied:
Removed

LukasPaczos commented on library/build.gradle:
```diff
@@ -1,4 +1,8 @@
 apply plugin: 'com.android.library'
+apply plugin: 'com.jfrog.bintray'
+apply plugin: "com.jfrog.artifactory"
+apply plugin: 'digital.wup.android-maven-publish'
```
Same as above.

langsmith replied:
Removed

LukasPaczos commented on library/build.gradle:
```diff
@@ -43,4 +47,5 @@ dependencies {
-apply from: 'javadoc.gradle'
\ No newline at end of file
+apply from: "${rootDir}/gradle/gradle-bintray.gradle"
+apply from: "${rootDir}/gradle/artifact-settings.gradle"
```
Same as above.

langsmith replied:
Removed
